CloudClub-uoft/iot-device
# Housekeeping

- Database

  - The Mongoose models are automatically bound to the default Mongoose connection from `mongoose.connect`, so it isn't necessary to pass around the connection.

  - Don't need to call `mongoose.model()` every time, it's already done in the schema we `require()`.

- Routes

  - `/device/info` should be using URL parameters instead of request body, since it is a GET request now. Also should be taking a MAC address.

  - `/device/unregister` can be shortened to one database query.

- Tests

  - Move test dependencies to `devDependencies`.

  - Rename `server` to `app` for consistency. If it should be `server` everywhere then let me know.

  - Take advantage of `chai`'s `expect` syntax for more readable tests.

  - Add GitHub workflow for running and confirming all tests passing on PR.

- Add `POST`, `GET`, `mochatest` VS Code snippets.

- Remove unused `express-router` dependency.

- ESLint should work properly now.
